pyjelly/integrations/rdflib/parse.py

```diff
@@ -366,97 +312,80 @@ def graphs_from_jelly(
+        yield parse_flat_triples_stream(frames=frames, options=options, store=store)  # type: ignore[call-arg]
         return
 
     if options.stream_types.logical_type == jelly.LOGICAL_STREAM_TYPE_FLAT_QUADS:
-        yield parse_flat_quads_stream(frames=frames, options=options, store=store)
+        yield parse_flat_quads_stream(frames=frames, options=options, store=store)  # type: ignore[call-arg]
         return
 
     if options.stream_types.logical_type == jelly.LOGICAL_STREAM_TYPE_GRAPHS:
-        yield from parse_graph_stream(frames=frames, options=options, store=store)
+        yield from parse_graph_stream(frames=frames, options=options, store=store)  # type: ignore[call-arg]
         return
 
     logical_type_name = jelly.LogicalStreamType.Name(options.stream_types.logical_type)
     msg = f"the stream type {logical_type_name} is not supported "
     raise NotImplementedError(msg)
 
 
-def graph_from_jelly(
+def parse_jelly_flat(
     inp: IO[bytes],
-    store: Store | str = "default",
-    identifier: str | None = None,
+    sink: Graph,
+    lambda_graph: Callable[[Any, Any], Graph],
+    lambda_dataset: Callable[[Any], Dataset],
 ) -> Any | Dataset | Graph:
     """
-    Parse graph from jelly.
+    Parse jelly file with FLAT physical type.
 
     Notes:
-        Compared to previous function, returns actual
-        graph/datasets.
+        Compared to previous function, modifies existing Graph/Dataset.
 
     Args:
         inp (IO[bytes]): input jelly buffered binary stream
-        store (Store | str, optional): RDFLib store. Defaults to "default".
-        identifier (str | None, optional): Graph identifier from RDFLib obj.
-            Defaults to None.
+        sink (Graph): RDFLib Graph
+        lambda_graph (Callable): lambda to construct a Graph
+        lambda_dataset (Callable): lambda to construct a Dataset
 
     Raises:
-        NotImplementedError: raises an error if multiple datasets are in the stream?
-            TODO: probably should not be a problem for logical-type agnostic parsing?
-        NotImplementedError: if logical type is not supported
+        NotImplementedError: if physical type is not supported
 
     Returns:
-        Any | Dataset | Graph: returns dataset if logical type is QUADS or GRAPHS
-            with passed identifier assigned to a default context
+        RDFLib Graph or Dataset
 
     """
     options, frames = get_options_and_frames(inp)
 
-    if options.stream_types.logical_type == jelly.LOGICAL_STREAM_TYPE_DATASETS:
-        msg = (
-            "the stream contains multiple datasets and cannot be parsed into "
-            "a single dataset"
-        )
-        raise NotImplementedError(msg)
-
-    if options.stream_types.logical_type == jelly.LOGICAL_STREAM_TYPE_FLAT_TRIPLES:
-        return parse_flat_triples_stream(
+    if options.stream_types.physical_type == jelly.PHYSICAL_STREAM_TYPE_TRIPLES:
+        graph = lambda_graph(sink.store, sink.identifier)
```

Comment:
And we are back to square one with this – we still use sink.store here, which is what we wanted to avoid.

The lambda must be parameterless. We do not pass anything to it. It's up to the caller to make the lambda capture variables that are needed for constructing the graph.